providers/unleash/pkg/provider.go

```diff
@@ -0,0 +1,168 @@
+package unleash
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Unleash/unleash-client-go/v3"
+	"github.com/Unleash/unleash-client-go/v3/api"
+	of "github.com/open-feature/go-sdk/pkg/openfeature"
+)
+
+const providerNotReady = "Provider not ready"
+const generalError = "general error"
+
+type Provider struct {
+	status      of.State
+	eventStream chan of.Event
+}
+
+func NewProvider(providerConfig ProviderConfig) (*Provider, error) {
+	return NewProviderWithContext(context.Background(), providerConfig)
+}
+
+func NewProviderWithContext(ctx context.Context, providerConfig ProviderConfig) (*Provider, error) {
+	if providerConfig.Options != nil {
+		unleash.Initialize(
+			providerConfig.Options...,
+		)
+	}
+
+	provider := &Provider{
+		status: of.ReadyState,
+	}
+
+	return provider, nil
+}
+
+func (p *Provider) Status() of.State {
```

Comment:
We can complete the `StateHandler` interface implementation by implementing `Init(evaluationContext EvaluationContext) error`

And the initialization can wrap the `unleash.Initialize(...)` , including handling errors.